Some cppcheks fixes

vitalii-komisarenko opened the pull request:
It is nice to have code that follows code style guidelines.

**To reviewers**: something may be wrong in this PR.
@RanRegev noted that using const& is not always the right thing and the tests are indeed failing from time to time.

vitalii-komisarenko committed:
cppcheck: bcl: fix "pass by const reference" issues

Some std::string parameters in bcl methods/subroutines were supplied by value.
According to current code style, they should be supplied by reference.

This commit fixes these issues if the solution is straightforward.

Some subroutines reuse supplied argument as a local variable -- these ones
are out of scope of this commit and left unfixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: bcl: remove "pass by const&" issues

"Pass by const reference" issues in bcl have been fixed,
so they can be removed from the list of known issues.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: update_vap_stats: pass args by const&

Pass `vap_iface_name` into `update_vap_stats` as `const std::string&` in all BWL versions.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: update_stations_stats: pass args by const&

Pass `update_stations_stats` string arguments as `const std::string&`
in all versions of BWL.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: sta_wlan_hal_create: pass args by const&

Pass `iface_name` into `sta_wlan_hal_create` as `const std::string&` in all BWL versions.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: sta_wlan_hal_*: pass string constructor args by const&

All version of sta_wlan_hal_* have been changed to accept `iface_name`
as `const std::string&` in the constructors.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: ap_wlan_hal_*: pass string constructor args by const&

All version of ap_wlan_hal_* have been changed to accept `iface_name`
as `const std::string&` in the constructors.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: mon_wlan_hal_*: pass string constructor args by const&

All version of mon_wlan_hal_* have been changed to accept `iface_name`
as `const std::string&` in the constructors.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
bwl: base_wlan_hal_*: pass string constructor args by const&

All version of base_wlan_hal_* have been changed to accept `iface_name`
as `const std::string&` in the constructors.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: bwl: remove "pass by const&" issues

Remove BWL "pass by const reference" from the list of known cppcheck issues
because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
platform_manager_thread: fix "pass by const&" cppcheck issues

Fix "pass by const reference" cppcheck issuese in platform_manager_thread.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: platform_manager_thread: remove "pass by const&"

Remove platform_manager_thread "pass by const reference" issues
from the list of known issues because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
backhaul_manager: wan_monitor: pass arg by const& to `initialize`

Pass `strWanIfaceName` parameter to `initialize` by const reference.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: wan_monitor: remove "pass by const&" issues

Remove "pass by const&" wan_monitor issue from the list of known issues
because it has been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
master: db: node: pass string args by const&

Pass std::string arguments by const reference in beerocks/master/db/node.{cpp.h}
in order to improve code style.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: master db node: remove "pass by const&"

Remove "pass by const reference" issues in beerocks/master/db/node.{cpp,h}
from the list of known cppcheck issues because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

RanRegev commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -451,7 +451,7 @@ static bool get_scan_results_from_nl_msg(sChannelScanResults &results, struct nl
 //////////////////////////////////////////////////////////////////////////////
 
-mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(std::string iface_name, hal_event_cb_t callback)
+mon_wlan_hal_dwpal::mon_wlan_hal_dwpal(const std::string &iface_name, hal_event_cb_t callback)
```
This constructor just as an example, this comment is for all changes that replaced a copy with a const reference:
are we sure the const reference we pass does not go out of scope leaving us with a dangling reference?
If we bind to a temporary, its lifetime is extended. But if we bind to a variable and this variable goes out of scope, then we have a problem.
reference: https://en.cppreference.com/w/cpp/language/reference#Dangling_references

vitalii-komisarenko replied:
I have to check it.
I hope there is no constructor/method that stores references to `std::string`s instead of copying them.

arnout replied:
In fact, if it was a reference to a std::string, it wouldn't have worked to begin with, because the temporary that was passed originally goes out of scope much quicker and it *will* be overwritten with something else on the stack. So if a reference was stored in the object, we'd already have noticed.

vitalii-komisarenko committed:
master: optimal_path_task: pass string args by const&

Pass std::string and std::set argument by const reference in optimal_path_task
in order to align better with the code style.

Virtual functions are out of scope of this commit,
because their changes need to be synchronized with the base class.
For the sake of simplicity they are not changed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

RanRegev reviewed (approved):
1. I have a general comment that I attached to one of the constructors.
2. I think it would be better to be more precise in the commit, namely, state that std::string is replaced with const std::string& and not the general "pass by const reference" case.

vitalii-komisarenko committed:
cppcheck_existing_issues: optimal_path_task: remove "pass by const&"

Remove optimal_path_task-related "pass by const reference" issues
from the list of known cppcheck issues because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

arnout reviewed (approved):
Excellent PR with good commit messages. There are a few mistakes in it though...
- some of the commit message say "bcl" while it's actually bwl;
- typo agrs -> args;
- the commit messages are sometimes worded slightly differently. It's better to always use the same wording, that's easier to read when you browse the logs. Use `git commit -c HEAD` to repeat the same commit message.

arnout commented:
Oh BTW no reason to keep this in draft. It's better to merge things as quickly as possible. Less risk of conflicts that way. Better two separate PRs than a single one that takes days to get merged.
